jax/_src/state/utils.py

```diff
@@ -29,7 +29,7 @@ def hoist_consts_to_refs(jaxpr: core.Jaxpr) -> core.Jaxpr:
   is_const_ref = [isinstance(var.aval, AbstractRef) for var in
                   jaxpr.constvars]
   const_avals_, const_ref_avals = partition_list(is_const_ref, all_const_avals)
-  const_avals = map(AbstractRef, const_avals_)
+  const_avals: list[AbstractRef] = list(unsafe_map(AbstractRef, const_avals_))
```

Comment:
The safe map ensures that the various argument lists have the same length, but there's only one argument list.  The safe map didn't seem to work well with type checking, so avoiding it.